src/AzureIoTHub.Portal.Infrastructure/Services/AWS/AWSExternalDeviceService.cs

```diff
@@ -53,6 +53,18 @@ public async Task<CreateThingResponse> CreateDevice(CreateThingRequest device)
             return thingResponse;
         }
 
+        public async Task<UpdateThingResponse> UpdateDevice(UpdateThingRequest device)
+        {
+            var thingResponse = await this.amazonIotClient.UpdateThingAsync(device);
+
+            if (thingResponse.HttpStatusCode != System.Net.HttpStatusCode.OK)
+            {
+                throw new InternalServerErrorException($"Unable to update the thing with id {device.ThingName} due to an error in the Amazon IoT API : {thingResponse.HttpStatusCode}");
```

Comment:
Message is confusing, is the device identifier or name?

Reply:
Well seen ! I updated the wrong messages